registrations/tests/test_send_message.py

```diff
@@ -0,0 +1,247 @@
+import pytest
+from django.conf import settings
+from django.core import mail
+from rest_framework import status
+
+from events.tests.utils import versioned_reverse as reverse
+
+
+def send_message(api_client, registration_id, send_message_data):
+    send_message_url = reverse(
+        "registration-send-message", kwargs={"pk": registration_id}
+    )
+    response = api_client.post(send_message_url, send_message_data, format="json")
+
+    return response
+
+
+def assert_send_message(
+    api_client, registration_id, send_message_data, expected_emails
+):
+    response = send_message(api_client, registration_id, send_message_data)
+
+    assert response.status_code == status.HTTP_200_OK
+    assert len(mail.outbox) == len(expected_emails)
+    valid_emails = [mail.to for mail in mail.outbox]
+    for idx, email in enumerate(expected_emails):
+        assert mail.outbox[idx].subject == send_message_data["subject"]
+        assert mail.outbox[idx].body == send_message_data["body"]
+        assert mail.outbox[idx].from_email == settings.SUPPORT_EMAIL
+        assert [email] in valid_emails
+
+
+@pytest.mark.django_db
+def test_admin_user_can_send_message_to_all_signups(
+    api_client, registration, signup, signup2, user
+):
+    api_client.force_authenticate(user)
+    send_message_data = {"subject": "Message subject", "body": "Message body"}
+
+    assert_send_message(
+        api_client, registration.id, send_message_data, [signup.email, signup2.email]
+    )
+
+
+@pytest.mark.parametrize(
+    "required_field",
+    [
+        "subject",
+        "body",
+    ],
+)
+@pytest.mark.django_db
+def test_required_fields_has_to_be_filled(
+    api_client, registration, required_field, signup, signup2, user
+):
+    api_client.force_authenticate(user)
+
+    send_message_data = {"subject": "Message subject", "body": "Message body"}
+
+    assert_send_message(
+        api_client, registration.id, send_message_data, [signup.email, signup2.email]
+    )
+
+    send_message_data[required_field] = ""
+
+    response = send_message(api_client, registration.id, send_message_data)
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert response.data[required_field][0].code == "blank"
+
+
+@pytest.mark.django_db
+def test_send_message_to_selected_signups(
+    api_client, registration, signup, signup2, user
+):
+    api_client.force_authenticate(user)
+    send_message_data = {
+        "subject": "Message subject",
+        "body": "Message body",
+        "signups": [signup.id],
```

Comment:
Might make sense to have multiple IDs here. Or separate tests/parameters for selected signups and a selected signup.

Reply:
Added another signup `[signup.id, signup3.id]` to this test to make sure that email is sent to multiple participants